R/public.R

```diff
@@ -0,0 +1,102 @@
+#' Format numbers for public consumption
+#'
+#' In public reports, cells are only reported if the N count is greater than
+#' some minimum number (usually 25). In addition, we:
+#'   * Round figures to some number of digits, retaining trailing 0s,
+#'     e.g. 25 -> 25.0
+#'   * Display confidence intervals in brackets beside estimates,
+#'     e.g. '25.0 (24.5, 25.5)'
+#'
+#' @param x A numeric vector, typically the 'face value' of a projectable `col`
+#' @param ci_lower A numeric vector, the lower bound of the confidence interval
+#' @param ci_upper A numeric vector, the upper bound of the confidence interval
+#' @param N A numeric vector, the base count
+#' @param digits A number, the number of digits to values to
+#' @param min_N A number, the minimum N required for values to be reportable
+#'
+#' @return A character vector
+#'
+#' @examples
+#'
+#' # Present a formatted number with suppressions applied:
+#' x <- projectable::col_freq(1:5, 1:5*10, summarised = TRUE)
+#' x <- projectable::prj_project_col(x)
+#' public_num(x$p, x$N, digits = 2)
+#'
+#' # Present a formatted number and its confidence interval with suppressions applied
+#' x <- projectable::col_binomial(1:5, 1:5*10, summarised = TRUE)
+#' x <- projectable::prj_project_col(x)
+#' public_ci(x$p, x$N, x$ci_lower, x$ci_upper)
+#'
+#' @name public_formats
+NULL
+
+#' @export
+#' @rdname public_formats
+public_num <- function(x, N, digits = 1, min_N = 25) {
```

Comment:
I'm still not entirely convinced these belong here. They're much more related to how SRC does reporting than the `projectable` class and associated methods. Also would be a bit cleaner to introduce these in a separate PR so that they have their own commit